clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -264,8 +292,7 @@ protected LogEntry makeNext() {
                         // which will de-compress the payload to a set of messages;
                         // since we assume nested compression is not allowed, the deep iterator
                         // would not try to further decompress underlying messages
```

Comment:
It would probably be good to mention that `innerIter` will always have at least one element (and hence why we can just call `next()` on it without calling `hasNext()` first.